BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -2335,12 +2362,21 @@ void AAMGuidance()
                 }
                 else// AAM Lead
                     aamTarget = MissileGuidance.GetAirToAirTarget(TargetPosition, TargetVelocity, TargetAcceleration, vessel, out timeToImpact, optimumAirspeed);
-
-
-                if (Vector3.Angle(aamTarget - transform.position, transform.forward) > maxOffBoresight * 0.75f)
+                if(hasDataLink && (vrd != null && vrd.locked) && (Vector3.Angle(aamTarget - transform.position, transform.forward) > maxOffBoresight * 0.75f))
                 {
-                    aamTarget = TargetPosition;
+                    if(Vector3.Angle(aamTarget - transform.position, transform.forward) > maxOffBoresight){
+                        TotalDrift += DataLinkDrift;
+                        aamTarget = MissileGuidance.GetDLDeviation(aamTarget,TotalDrift);
```

Comment:
This doesn't make much sense to me here, drift from an INS perspective should be drift of the missile's positional data, not the target's positional data. This should be affecting the guidance calculations themselves.